src/test/java/org/jabref/logic/importer/fetcher/ZbMATHTest.java

```diff
@@ -44,18 +43,42 @@ void setUp() throws Exception {
         donaldsonEntry.setField(StandardField.TITLE, "An application of gauge theory to four dimensional topology");
         donaldsonEntry.setField(StandardField.VOLUME, "18");
         donaldsonEntry.setField(StandardField.YEAR, "1983");
-        donaldsonEntry.setField(new UnknownField("zbl"), "0507.57010");
+        donaldsonEntry.setField(StandardField.ZBL_NUMBER, "0507.57010");
     }
 
     @Test
     void searchByQueryFindsEntry() throws Exception {
         List<BibEntry> fetchedEntries = fetcher.performSearch("an:0507.57010");
-        assertEquals(Collections.singletonList(donaldsonEntry), fetchedEntries);
+        assertEquals(donaldsonEntry, fetchedEntries.get(0));
     }
 
     @Test
     void searchByIdFindsEntry() throws Exception {
         Optional<BibEntry> fetchedEntry = fetcher.performSearchById("0507.57010");
         assertEquals(Optional.of(donaldsonEntry), fetchedEntry);
     }
+
+    @Test
+    void searchByEntryFindsEntry() throws Exception {
+        BibEntry searchEntry = new BibEntry();
+        searchEntry.setField(StandardField.TITLE, "An application of gauge theory to four dimensional topology");
+        searchEntry.setField(StandardField.AUTHOR, "S. K. {Donaldson}");
+
+        List<BibEntry> fetchedEntries = fetcher.performSearch(searchEntry);
+        assertEquals(donaldsonEntry, fetchedEntries.get(0));
```

Comment:
As above, I would prefer if that would compare against `Collections.singletonList(donaldsonEntry)`